validator/validate.py

```diff
@@ -0,0 +1,80 @@
+"""Configuration files validator"""
+
+import os
+from pathlib import Path
+
+import jsonschema
+import yaml
+
+from validator.config.config import Configuration, SupportedValidations
+from validator.errors.errors import InvalidTargetPathError
+from validator.schemas.schema import SCHEMAS
+
+
+def build_app_path():
+    """Builds the path for /apps folder"""
+    target_path = Configuration.get_target_path()
+    if target_path is None:
+        raise InvalidTargetPathError(target_path)
+    return os.path.join(target_path, "apps")
+
+
+def build_project_conf_path():
+    """Builds the path for project.yaml"""
+    target_path = Configuration.get_target_path()
+    if target_path is None:
+        raise InvalidTargetPathError(target_path)
+    return os.path.join(target_path, "project.yaml")
+
+
+def validate_file(file_path: str, schema: dict) -> str:
+    """Validates a file with a given schema
+
+    Args:
+        file_path (string): The path to the file
+        schema (Dict): The schema format
+
+    Returns:
+        String: The validated file
+    """
+    with open(file_path, "r") as file:
+        yaml_file_data = file.read()
+        yaml_file = yaml.safe_load(yaml_file_data)
+        return jsonschema.validate(yaml_file, schema)
+
+
+def validate():
```

Comment:
I would change the prototype to have passed the validation target and the file to validate to have something like
```suggestion
def validate(file_to_validate, target_validation):
```

This way you will be able to call this in the config lib 🙂 
So it means rewriting a bit this function but it will way easier to call afterwards.

Reply:
There's already a `validate_file` function in this file, just above that one. One validates file and this one validates paths, I will rename it to make it more specific 👍